services/cronback-project-srv/handler.rs

```diff
@@ -0,0 +1,176 @@
+    ) -> Result<Response<CreateProjectResponse>, Status> {
+        let id = ProjectId::generate();
+        let project = Project {
+            id: id.clone(),
+            created_at: Utc::now(),
+            last_status_changed_at: Utc::now(),
+            status: ProjectStatus::Enabled,
+        };
+
+        self.project_store
+            .store_project(project)
+            .await
+            .map_err(ProjectStoreHandlerError::Store)?;
+
+        e!(project_id = id, ProjectCreated {});
+
+        Ok(Response::new(CreateProjectResponse {
+            id: Some(id.into_inner().into()),
+        }))
+    }
+
+    async fn get_project_status(
+        &self,
+        request: Request<GetProjectStatusRequest>,
+    ) -> Result<Response<GetProjectStatusResponse>, Status> {
+        let (_, _, req) = request.into_parts();
+        let project_id: ProjectId = req.id.unwrap().into();
+        let project_id = project_id
+            .validated()
+            .map_err(|e| Status::invalid_argument(e.to_string()))?;
+        let status = self
+            .project_store
+            .get_status(&project_id)
+            .await
+            .map_err(ProjectStoreHandlerError::Store)?;
+        match status {
+            | Some(st) => {
+                Ok(Response::new(GetProjectStatusResponse {
+                    status: st.into(),
+                }))
+            }
+            | None => {
+                Err(ProjectStoreHandlerError::NotFound(format!(
+                    "{}",
+                    project_id
+                )))?
+            }
+        }
+    }
+
+    async fn set_project_status(
+        &self,
+        request: Request<SetProjectStatusRequest>,
+    ) -> Result<Response<SetProjectStatusResponse>, Status> {
+        let (_, _, req) = request.into_parts();
+        let project_id: ProjectId = req.id.unwrap().into();
+        let project_id = project_id
+            .validated()
+            .map_err(|e| Status::invalid_argument(e.to_string()))?;
+
+        let old_status = self
+            .project_store
+            .get_status(&project_id)
+            .await
+            .map_err(ProjectStoreHandlerError::Store)?;
+
+        let Some(old_status) = old_status else {
+            return Err(ProjectStoreHandlerError::NotFound(
+                project_id.to_string(),
+            )
+            .into());
+        };
+
+        self.project_store
+            .set_status(&project_id, req.status.into())
+            .await
+            .map_err(ProjectStoreHandlerError::Store)?;
+
+        // NOTE: The `old_status` here might not always be 100% accurate as it
+        // might race with another update.
+        e!(
+            project_id = project_id,
+            ProjectStatusUpdated {
+                old_status: old_status.into(),
+                new_status: req.status,
+            }
+        );
+
+        Ok(Response::new(SetProjectStatusResponse {
+            status: req.status,
```

Comment:
I've been assuming that we return the old_status instead of the new input. Isn't this the convention for many set/update operations?